src/main/frontend/db/query_react.cljs

```diff
@@ -16,40 +16,56 @@
             [lambdaisland.glogi :as log]))
 
 (defn resolve-input
-  [input]
-  (cond
-    (= :right-now-ms input) (util/time-ms)
-    (= :start-of-today-ms input) (util/today-at-local-ms 0 0 0 0)
-    (= :end-of-today-ms input) (util/today-at-local-ms 24 0 0 0)
+  ([input]
+   (resolve-input input nil))
+  ([input current-block-uuid]
+   (cond
+     (= :right-now-ms input) (util/time-ms)
+     (= :start-of-today-ms input) (util/date-at-local-ms 0 0 0 0)
+     (= :end-of-today-ms input) (util/date-at-local-ms 24 0 0 0)
 
-    (= :today input)
-    (date->int (t/today))
-    (= :yesterday input)
-    (date->int (t/minus (t/today) (t/days 1)))
-    (= :tomorrow input)
-    (date->int (t/plus (t/today) (t/days 1)))
-    (= :current-page input)
-    (some-> (or (state/get-current-page)
-                (:page (state/get-default-home))
-                (date/today)) string/lower-case)
+     (= :today input)
+     (date->int (t/today))
+     (= :yesterday input)
+     (date->int (t/minus (t/today) (t/days 1)))
+     (= :tomorrow input)
+     (date->int (t/plus (t/today) (t/days 1)))
+     (= :current-page input)
+     (some-> (or (state/get-current-page)
+                 (:page (state/get-default-home))
+                 (date/today)) string/lower-case)
+     (= :current-block input)
+     (:db/id (db-utils/entity [:block/uuid current-block-uuid]))
+     (= :parent-block input)
+     (:db/id (model/get-block-parent current-block-uuid))
 
-    (and (keyword? input)
-         (util/safe-re-find #"^\d+d(-before)?$" (name input)))
-    (let [input (name input)
-          days (parse-long (re-find #"^\d+" input))]
-      (date->int (t/minus (t/today) (t/days days))))
-    (and (keyword? input)
-         (util/safe-re-find #"^\d+d(-after)?$" (name input)))
-    (let [input (name input)
-          days (parse-long (re-find #"^\d+" input))]
-      (date->int (t/plus (t/today) (t/days days))))
+     (and (keyword? input)
+          (util/safe-re-find #"^\d+d(-before-ms)?$" (name input)))
```

Comment:
We can just use `re-find` here because the keyword check guarantees the second argument is going to be a string